src/geojson.rs

```diff
@@ -271,4 +295,29 @@ mod tests {
             })
         );
     }
+
+    #[test]
+    fn test_invalid_json() {
+        let geojson_str = r#"{
+           "type": "FeatureCollection",
+           "features": [
+             !INTENTIONAL_TYPO! {
+               "type": "Feature",
+               "properties": {},
+               "geometry": {
+                 "type": "Point",
+                 "coordinates": [
+                   -0.13583511114120483,
+                   51.5218870403801
+                 ]
+               }
+             }
+           ]
+        }"#;
+        match GeoJson::from_str(geojson_str) {
+            Ok(_) => panic!("unexpectedly succeeded at parsing invalid json"),
+            Err(Error::MalformedJson(_)) => (), // expected
+            Err(e) => panic!("unexpected error: {}", e),
+        }
```

Comment:
i haven't used it before, but i think you can use [`matches!`](https://doc.rust-lang.org/std/macro.matches.html) here. not a blocker to merging

```suggestion
        assert!(
          matches!(GeoJson::from_str(geojson_str), Err(Error::MalformedJson(_)))
        );
```

Reply:
Done!

I'm glad to learn this exists - it felt pretty clunky as it was. Thanks @frewsxcv!

Reply:
TIL